app/assets/javascripts/analytics-ga4/ga4-finder-tracker.js

```diff
@@ -152,6 +152,24 @@
       } catch (e) {
         console.error('GA4 configuration error: ' + e.message, window.location)
       }
+    },
+
+    addFilterButtonTracking: function (button, section) {
+      button.setAttribute('data-ga4-expandable', '')
+      var ga4JSON = { event_name: 'select_content', type: 'finder', section: section }
+
+      try {
+        var index = button.closest('[data-ga4-index]') || undefined
+        if (index) {
+          ga4JSON.index = JSON.parse(index.getAttribute('data-ga4-index'))
+        }
+      } catch (e) {
+        // if there's a problem with the index object, don't add the attribute
+        console.error('GA4 configuration error: ' + e.message, window.location)
+        return
+      }
+
+      button.setAttribute('data-ga4-event', JSON.stringify(ga4JSON))
```

Comment:
From my reading of this, if it doesn't find a parent with `data-ga4-index` this line will still run, but with the default JSON above and not including any index data. Is that okay, does the code that handles this later deal with it, or do we need to put something in here to flag that something's gone wrong in this situation?

Reply:
Thanks, I've added some `console.warn` statements if the index doesn't exist. Let me know if you think we should do something different instead.